exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet2/DrillParquetReader.java

```diff
@@ -179,6 +189,31 @@ private static MessageType getProjection(MessageType schema,
     return projection;
   }
 
+  /**
+   * Get type from the supplied {@code type} corresponding to given {@code segment}.
+   * @param type type to extract field corresponding to segment
+   * @param segment segment which type will be returned
+   * @return type corresponding to the {@code segment} or {@code null} if there is no field found in {@code type}.
+   */
+  private static Type getType(Type type, PathSegment segment) {
+    Type result = null;
+    if (type != null && !type.isPrimitive()) {
+      GroupType groupType = type.asGroupType();
+      if (segment.isNamed()) {
+        String fieldName = segment.getNameSegment().getPath();
+        Optional<Type> foundType = groupType.getFields().stream()
+            .filter(f -> f.getName().equalsIgnoreCase(fieldName))
+            .findAny();
+        result = foundType.map(field -> groupType.getType(field.getName()))
+            .orElse(null);
+      } else if (type.getOriginalType() == OriginalType.LIST) { // the segment is array index
+        // get element type of the list
+        result = groupType.getType(0).asGroupType().getType(0);
```

Comment:
Unsafe when list's element type isn't group. Such schema does not meet the logical list definition, but the case is possible.